demo/KevinDemo/Extensions/String+Extensions.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  String+Extensions.swift
+//  KevinDemo
+//
+//  Created by Daniel Klinge on 21/02/2022.
+//
+
+import Foundation
+
+extension String {
+    func isValidEmail() -> Bool {
+        if self.count > 100 {
+            return false
+        }
+
+        let emailFormat = "(?:[\\p{L}0-9!#$%\\&'*+/=?\\^_`{|}~-]+(?:\\.[\\p{L}0-9!#$%\\&'*+/=?\\^_`{|}" + "~-]+)*|\"(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21\\x23-\\x5b\\x5d-\\" + "x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])*\")@(?:(?:[\\p{L}0-9](?:[a-" + "z0-9-]*[\\p{L}0-9])?\\.)+[\\p{L}0-9](?:[\\p{L}0-9-]*[\\p{L}0-9])?|\\[(?:(?:25[0-5" + "]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-" + "9][0-9]?|[\\p{L}0-9-]*[\\p{L}0-9]:(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21" + "-\\x5a\\x53-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])+)\\])"
```

Comment:
This one is better:

```swift
public func isValidEmail() -> Bool {
    let emailFormat =
        "^[_A-Za-z0-9-+]+(\\.[_A-Za-z0-9-+]+)*@[A-Za-z0-9-]+(\\.[A-Za-z0-9-]+)*(\\.[A-Za-z‌​]{2,})$"
        
    let regex = try? Regex(
        string: emailFormat,
        options: [.ignoreCase]
    )
        
    return regex?.matches(self) ?? false
}
```

Reply:
I assume code uses some sort of Regex package (maybe https://swiftpack.co/package/sharplet/Regex)? Because natively Swift doesn't know what Regex is - it operates with predicates.